File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
    let n = extraCandies;
    let tmp = [];
    var arrSorted = candies.toSorted((a,b) => b-a);
    const maxn = arrSorted[0];
    for(var x of candies){
        if(x+n>=maxn) tmp.push(true);
        else tmp.push(false)
    }
    return tmp
};

File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        n = extraCandies
        tmp = max(candies)
        result = []
        for x in candies : 
            if x + n >= tmp:
                result.append(True)
            else :
                result.append(False)
        return result